### ISSUE TITLE
Construction for FiniteRankDualFreeModule

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#34605]
<div id="comment:0"></div>

... using `DualFunctor`.

Depends on #34448

CC:  @egourgoulhon @tscrim

Component: **linear algebra**

Branch/Commit: **[u/mkoeppe/construction_for_finiterankdualfreemodule](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/construction_for_finiterankdualfreemodule) @ [`cba3a87`](https://github.com/sagemath/sagetrac-mirror/commit/cba3a8730c1f9cdb2f5dd1a6672fdb27163cba85)**

_Issue created by migration from https://trac.sagemath.org/ticket/34605_

**Comment by mkoeppe** [comment]
Dependencies: **#34448**

**Comment by mkoeppe** [comment]
Branch: **[u/mkoeppe/construction_for_finiterankdualfreemodule](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/construction_for_finiterankdualfreemodule)**

**Comment by mkoeppe** [comment]
Commit: **[`cba3a87`](https://github.com/sagemath/sagetrac-mirror/commit/cba3a8730c1f9cdb2f5dd1a6672fdb27163cba85)**

**Comment by mkoeppe** [comment]
<div id="comment:3" align="right">comment:3</div>

Pushout does not work yet; I think `DualFunctor` will have to be promoted from `Functor` to `ConstructionFunctor` for this

---
Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/06710c12cc1d73cafe81e087742a514808985f86"><code>06710c1</code></a></td><td><code>src/sage/manifolds/differentiable/diff_form_module.py, vectorfield_module.py: Update copyright using 'git blame -w --date=format:%Y FILE.py | sort -k2', add to AUTHORS</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/bb614c17c7c9d396be1376b526c5eb52c39cba7d"><code>bb614c1</code></a></td><td><code>Merge #34501</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/4a4a29d6cd7d49d4ea9c1720566ce50ba8f4607c"><code>4a4a29d</code></a></td><td><code>Merge tag '9.7' into t/34448/put_tensor_modules_of_finiterankfreemodule_in_modules___tensorproducts__</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1f4fb90a273b81a7ea9cc14f97c7842eacd022a8"><code>1f4fb90</code></a></td><td><code>Merge tag '9.8.beta0' into t/34448/put_tensor_modules_of_finiterankfreemodule_in_modules___tensorproducts__</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7c0495988c7962163abf3c11ad200add7feb05cc"><code>7c04959</code></a></td><td><code>src/sage/manifolds/differentiable/diff_form_module.py: Fix typo in docstring</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/fc31cbf584a3de4ffa7005c33a4b8ef4a98d09ca"><code>fc31cbf</code></a></td><td><code>Merge #34501</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f0def924fea5ab8caaeccd0f89d21d455004a6aa"><code>f0def92</code></a></td><td><code>src/sage/tensor/modules/tensor_free_module.py: Remove redundant method</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e10f9cf17bd06784805914b63b7ae0adc4423c96"><code>e10f9cf</code></a></td><td><code>src/sage/manifolds/differentiable/vectorfield_module.py: Revert change to VectorFieldModule.tensor (not needed yet)</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/f947c1811b7c601f8b27cb903f0140c6c0ca9ed3"><code>f947c18</code></a></td><td><code>Merge #34448</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/cba3a8730c1f9cdb2f5dd1a6672fdb27163cba85"><code>cba3a87</code></a></td><td><code>FiniteRankDualFreeModule.construction: WIP</code></td></tr></table>
